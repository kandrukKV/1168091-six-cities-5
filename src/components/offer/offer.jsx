import React from "react";
import {NavLink} from "react-router-dom";
import PropTypes from "prop-types";
import Rating from "../rating/rating";
import PremiumLabel from "../premium-label/premium-label";
import Additions from "../additions/additions";
import Reviews from "../reviews/reviews";
import ReviewForm from "../review-form/review-form";
import MeetTheHost from "../meet-the-host/meet-the-host";

const Offer = (props) => {

  const {card, reviews} = props;

  return (
    <div className="page">
      <header className="header">
        <div className="container">
          <div className="header__wrapper">
            <div className="header__left">
              <NavLink to="/" className="header__logo-link">
                <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
              </NavLink>
            </div>
            <nav className="header__nav">
              <ul className="header__nav-list">
                <li className="header__nav-item user">
                  <NavLink to="/favorites" className="header__nav-link header__nav-link--profile">
                    <div className="header__avatar-wrapper user__avatar-wrapper">
                    </div>
                    <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                  </NavLink>
                </li>
              </ul>
            </nav>
          </div>
        </div>
      </header>

      <main className="page__main page__main--property">
        <section className="property">
          <div className="property__gallery-container container">
            <div className="property__gallery">
              {
                card.photos.map((photo, i) => {
                  return (
                    <div className="property__image-wrapper" key={`photo-${i}`}>
                      <img className="property__image" src={photo.src} alt={photo.alt}/>
                    </div>
                  );
                })
              }
            </div>
          </div>
          <div className="property__container container">
            <div className="property__wrapper">
              {card.isPremium && <PremiumLabel className={`property`}/>}
              <div className="property__name-wrapper">
                <h1 className="property__name">
                  {card.title}
                </h1>
                <button className="property__bookmark-button button" type="button">
                  <svg className="property__bookmark-icon" width="31" height="33">
                    <use xlinkHref="#icon-bookmark"></use>
                  </svg>
                  <span className="visually-hidden">To bookmarks</span>
                </button>
              </div>

              <Rating rating={card.rating} className={`property`}/>

              <ul className="property__features">
                <li className="property__feature property__feature--entire">
                  {card.type}
                </li>
                <li className="property__feature property__feature--bedrooms">
                  {card.badrooms} Bedrooms
                </li>
                <li className="property__feature property__feature--adults">
                  Max {card.adults} adults
                </li>
              </ul>
              <div className="property__price">
                <b className="property__price-value">&euro;{card.price}</b>
                <span className="property__price-text">&nbsp;night</span>
              </div>
              <div className="property__inside">
                <h2 className="property__inside-title">What&apos;s inside</h2>
                <Additions additions={card.additions}/>
              </div>
              <MeetTheHost/>
              <section className="property__reviews reviews">
                <Reviews reviews={reviews}/>
                <ReviewForm/>
              </section>
            </div>
          </div>
          <section className="property__map map"></section>
        </section>
        <div className="container">
          <section className="near-places places">
            <h2 className="near-places__title">Other places in the neighbourhood</h2>
            <div className="near-places__list places__list">
              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/room.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;80</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button place-card__bookmark-button--active button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">In bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={ {width: `80%`} }></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Wood and stone place</a>
                  </h2>
                  <p className="place-card__type">Private room</p>
                </div>
              </article>

              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/apartment-02.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;132</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={ {width: `80%`} }></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Canal View Prinsengracht</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>

              <article className="near-places__card place-card">
                <div className="near-places__image-wrapper place-card__image-wrapper">
                  <a href="#">
                    <img className="place-card__image" src="img/apartment-03.jpg" width="260" height="200" alt="Place image"/>
                  </a>
                </div>
                <div className="place-card__info">
                  <div className="place-card__price-wrapper">
                    <div className="place-card__price">
                      <b className="place-card__price-value">&euro;180</b>
                      <span className="place-card__price-text">&#47;&nbsp;night</span>
                    </div>
                    <button className="place-card__bookmark-button button" type="button">
                      <svg className="place-card__bookmark-icon" width="18" height="19">
                        <use xlinkHref="#icon-bookmark"></use>
                      </svg>
                      <span className="visually-hidden">To bookmarks</span>
                    </button>
                  </div>
                  <div className="place-card__rating rating">
                    <div className="place-card__stars rating__stars">
                      <span style={ {width: `100%`} }></span>
                      <span className="visually-hidden">Rating</span>
                    </div>
                  </div>
                  <h2 className="place-card__name">
                    <a href="#">Nice, cozy, warm big bed apartment</a>
                  </h2>
                  <p className="place-card__type">Apartment</p>
                </div>
              </article>
            </div>
          </section>
        </div>
      </main>
    </div>
  );
};

Offer.propTypes = {
  card: PropTypes.shape({
    id: PropTypes.number.isRequired,
    city: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    photos: PropTypes.arrayOf(PropTypes.shape({
      src: PropTypes.string.isRequired,
      alt: PropTypes.string
    })),
    badrooms: PropTypes.number.isRequired,
    adults: PropTypes.number.isRequired,
    additions: PropTypes.array.isRequired,
    owner: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      avatar: PropTypes.string.isRequired
    }),
    description: PropTypes.string.isRequired
  }),
  reviews: PropTypes.array.isRequired
};

export default Offer;
