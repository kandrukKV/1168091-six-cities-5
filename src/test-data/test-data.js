export const noon = () => {};

export const additions = [
  `addition1`,
  `addition2`,
  `addition3`,
  `addition4`,
  `addition5`,
];

export const cities = [
  `Amsterdam`,
  `Paris`,
];

export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: false,
    rating: 1,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.902976,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 2,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.388540000000006,
      longitude: 4.899976,
      zoom: 16
    },
    id: 2
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: false,
    isPremium: true,
    rating: 3,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.364540000000005,
      longitude: 4.9019759999999994,
      zoom: 16
    },
    id: 3
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: false,
    rating: 4,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.385540000000006,
      longitude: 4.886976,
      zoom: 16
    },
    id: 4
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.37454,
        longitude: 4.897976,
        zoom: 13
      }
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg`,
    images: [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg`,
    ],
    title: `Perfectly located Castro`,
    isFavorite: true,
    isPremium: true,
    rating: 5,
    type: `house`,
    bedrooms: 3,
    adults: 7,
    price: 440,
    additions: [
      `Dishwasher`,
      `Dishwasher`,
      `Dishwasher`,
    ],
    host: {
      id: 25,
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`
    },
    description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
    location: {
      latitude: 52.397540000000006,
      longitude: 4.9099759999999995,
      zoom: 16
    },
    id: 5
  }
];

export const mosckReviews = [
  {
    id: 1,
    user: {
      id: 18,
      name: `Sophie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg`
    },
    rating: 3,
    comment: `We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)`,
    date: `2020-11-02T13:38:44.752Z`
  },
  {
    id: 2,
    user: {
      id: 16,
      name: `Mollie`,
      avatar: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg`
    },
    rating: 2,
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-11-02T13:38:44.752Z`
  }
];
