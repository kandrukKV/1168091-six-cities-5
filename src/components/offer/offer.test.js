import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import {Offer} from "./offer";
import {mockOffers, mosckReviews, noon} from "../../test-data/test-data";

it(`Offer is rendered correctly with offers`, () => {
  const renderer = new ShallowRenderer();
  const tree = renderer.render(
      <Offer
        id={1}
        card={mockOffers[0]}
        reviews={mosckReviews}
        nearPlaces={mockOffers}
        getCurrentOffer={noon}
        removeActiveCard={noon}
      />
  );
  expect(tree).toMatchSnapshot();
});
