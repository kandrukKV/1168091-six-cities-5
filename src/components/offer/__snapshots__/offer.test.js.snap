// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offer is rendered correctly with offers 1`] = `
<div
  className="page"
>
  <Memo(Connect(Header)) />
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <OfferDetails
        card={
          Object {
            "additions": Array [
              "Dishwasher",
              "Dishwasher",
              "Dishwasher",
            ],
            "adults": 7,
            "bedrooms": 3,
            "city": Object {
              "location": Object {
                "latitude": 52.37454,
                "longitude": 4.897976,
                "zoom": 13,
              },
              "name": "Amsterdam",
            },
            "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
            "host": Object {
              "avatar": "img/avatar-angelina.jpg",
              "id": 25,
              "name": "Angelina",
            },
            "id": 1,
            "images": Array [
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
            ],
            "isFavorite": false,
            "isPremium": false,
            "location": Object {
              "latitude": 52.385540000000006,
              "longitude": 4.902976,
              "zoom": 16,
            },
            "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
            "price": 440,
            "rating": 1,
            "title": "Perfectly located Castro",
            "type": "house",
          }
        }
        reviews={
          Array [
            Object {
              "comment": "We loved it so much, the house, the veiw, the location just great.. Highly reccomend :)",
              "date": "2020-11-02T13:38:44.752Z",
              "id": 1,
              "rating": 3,
              "user": Object {
                "avatar": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/9.jpg",
                "id": 18,
                "name": "Sophie",
              },
            },
            Object {
              "comment": "What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!",
              "date": "2020-11-02T13:38:44.752Z",
              "id": 2,
              "rating": 2,
              "user": Object {
                "avatar": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/7.jpg",
                "id": 16,
                "name": "Mollie",
              },
            },
          ]
        }
      />
      <Memo(Connect(Map))
        cards={
          Array [
            Object {
              "additions": Array [
                "Dishwasher",
                "Dishwasher",
                "Dishwasher",
              ],
              "adults": 7,
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 52.385540000000006,
                "longitude": 4.902976,
                "zoom": 16,
              },
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "price": 440,
              "rating": 1,
              "title": "Perfectly located Castro",
              "type": "house",
            },
            Object {
              "additions": Array [
                "Dishwasher",
                "Dishwasher",
                "Dishwasher",
              ],
              "adults": 7,
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 52.388540000000006,
                "longitude": 4.899976,
                "zoom": 16,
              },
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "price": 440,
              "rating": 2,
              "title": "Perfectly located Castro",
              "type": "house",
            },
            Object {
              "additions": Array [
                "Dishwasher",
                "Dishwasher",
                "Dishwasher",
              ],
              "adults": 7,
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "name": "Angelina",
              },
              "id": 3,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": false,
              "isPremium": true,
              "location": Object {
                "latitude": 52.364540000000005,
                "longitude": 4.9019759999999994,
                "zoom": 16,
              },
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "price": 440,
              "rating": 3,
              "title": "Perfectly located Castro",
              "type": "house",
            },
            Object {
              "additions": Array [
                "Dishwasher",
                "Dishwasher",
                "Dishwasher",
              ],
              "adults": 7,
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "name": "Angelina",
              },
              "id": 4,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": true,
              "isPremium": false,
              "location": Object {
                "latitude": 52.385540000000006,
                "longitude": 4.886976,
                "zoom": 16,
              },
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "price": 440,
              "rating": 4,
              "title": "Perfectly located Castro",
              "type": "house",
            },
            Object {
              "additions": Array [
                "Dishwasher",
                "Dishwasher",
                "Dishwasher",
              ],
              "adults": 7,
              "bedrooms": 3,
              "city": Object {
                "location": Object {
                  "latitude": 52.37454,
                  "longitude": 4.897976,
                  "zoom": 13,
                },
                "name": "Amsterdam",
              },
              "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
              "host": Object {
                "avatar": "img/avatar-angelina.jpg",
                "id": 25,
                "name": "Angelina",
              },
              "id": 5,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 52.397540000000006,
                "longitude": 4.9099759999999995,
                "zoom": 16,
              },
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              "price": 440,
              "rating": 5,
              "title": "Perfectly located Castro",
              "type": "house",
            },
          ]
        }
        className="property"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <Memo(Connect(PlaceCardList))
          cards={
            Array [
              Object {
                "additions": Array [
                  "Dishwasher",
                  "Dishwasher",
                  "Dishwasher",
                ],
                "adults": 7,
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "name": "Angelina",
                },
                "id": 1,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": false,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.385540000000006,
                  "longitude": 4.902976,
                  "zoom": 16,
                },
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "price": 440,
                "rating": 1,
                "title": "Perfectly located Castro",
                "type": "house",
              },
              Object {
                "additions": Array [
                  "Dishwasher",
                  "Dishwasher",
                  "Dishwasher",
                ],
                "adults": 7,
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "name": "Angelina",
                },
                "id": 2,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 52.388540000000006,
                  "longitude": 4.899976,
                  "zoom": 16,
                },
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "price": 440,
                "rating": 2,
                "title": "Perfectly located Castro",
                "type": "house",
              },
              Object {
                "additions": Array [
                  "Dishwasher",
                  "Dishwasher",
                  "Dishwasher",
                ],
                "adults": 7,
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "name": "Angelina",
                },
                "id": 3,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": false,
                "isPremium": true,
                "location": Object {
                  "latitude": 52.364540000000005,
                  "longitude": 4.9019759999999994,
                  "zoom": 16,
                },
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "price": 440,
                "rating": 3,
                "title": "Perfectly located Castro",
                "type": "house",
              },
              Object {
                "additions": Array [
                  "Dishwasher",
                  "Dishwasher",
                  "Dishwasher",
                ],
                "adults": 7,
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "name": "Angelina",
                },
                "id": 4,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": true,
                "isPremium": false,
                "location": Object {
                  "latitude": 52.385540000000006,
                  "longitude": 4.886976,
                  "zoom": 16,
                },
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "price": 440,
                "rating": 4,
                "title": "Perfectly located Castro",
                "type": "house",
              },
              Object {
                "additions": Array [
                  "Dishwasher",
                  "Dishwasher",
                  "Dishwasher",
                ],
                "adults": 7,
                "bedrooms": 3,
                "city": Object {
                  "location": Object {
                    "latitude": 52.37454,
                    "longitude": 4.897976,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                },
                "description": "Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.",
                "host": Object {
                  "avatar": "img/avatar-angelina.jpg",
                  "id": 25,
                  "name": "Angelina",
                },
                "id": 5,
                "images": Array [
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                  "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
                ],
                "isFavorite": true,
                "isPremium": true,
                "location": Object {
                  "latitude": 52.397540000000006,
                  "longitude": 4.9099759999999995,
                  "zoom": 16,
                },
                "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
                "price": 440,
                "rating": 5,
                "title": "Perfectly located Castro",
                "type": "house",
              },
            ]
          }
          itemClassName="near-places__card"
          listClassName="near-places__list"
          wrapClassName="near-places"
        />
      </section>
    </div>
  </main>
</div>
`;
